### PR TITLE
Patches to run Catalyst on FreeBSD

### PR DESCRIPTION
**Pull request by monsieurp**
Please someone from @gentoo/releng have a look. Thanks!

### CODE REVIEW

**Comment by dol-sen** [comment]
How are these patches different than Yuta's patches in bug #363577?
These don't give him any credit that I've seen.
I had already looked over the patches and was intending on applying/testing them this weekend.
I did want to move where the sed  command was determined in main.py.  Minor issue I was going to do when I applied them.

**Comment by monsieurp** [comment]
@dol-sen I've taken his patches and filed a PR instead to get them merged ASAP. I can give him credit in the body of each commit if you want to.

**Comment by monsieurp** [comment]
@dol-sen credits to Yuta added. I've got to add the tmpfs patch now.

File: catalyst/main.py [hunk 1]
```diff
@@ -430,13 +436,16 @@ def _main(parser, opts):
 		# catalyst cannot be run as a normal user due to chroots, mounts, etc
 		log.critical('This script requires root privileges to operate')
 
+       if os.uname().sysname == 'Linux':
```
Comment (robbat2) [inline catalyst/main.py]: Can you please specifically comment here that FreeBSD doesn't have namespaces at this time. If it got namespaces in future, you might want to improve this code.

**Comment by monsieurp** [comment]
@robbat2 fixed

**Comment by dol-sen** [comment]
Please close.  These patches have been pushed to the master branch with some modifications

**Comment by monsieurp** [comment]
Alright. Thanks!

**Pull request closed without merging**